proton-c/bindings/go/src/qpid.apache.org/proton/handlers.go

```diff
@@ -308,7 +316,7 @@ func (d *MessagingAdapter) HandleEvent(e Event) {
 			d,
 		}
 		if d.Prefetch > 0 {
-			d.flowcontroller = cHandler{C.pn_flowcontroller(C.int(d.Prefetch))}
+			d.flowcontroller = flowcontroller{ window:d.Prefetch, drained:0 }
```

Comment:
Go cases do not fall through by default so your first 3 cases do nothing. I think you mean:

switch e.Type() {
case ELinkLocalOpen, ELinkRemoteOpen, ELinkFlow, Edelivery: ...

Otherwise it is  a thing of beauty, should have done this ages ago. :)